diff --git a/plugins/linux-lockdown/fu-linux-lockdown-plugin.c b/plugins/linux-lockdown/fu-linux-lockdown-plugin.c
index 6c8d3ee94..ea8b85597 100644
--- a/plugins/linux-lockdown/fu-linux-lockdown-plugin.c
+++ b/plugins/linux-lockdown/fu-linux-lockdown-plugin.c
@@ -155,7 +155,7 @@ fu_linux_lockdown_plugin_add_security_attrs(FuPlugin *plugin, FuSecurityAttrs *a
 		g_info("failed to ensure attribute fix flags: %s", error_local->message);

 	if (self->lockdown == FU_LINUX_LOCKDOWN_UNKNOWN) {
-		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_MISSING_DATA);
+		fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_NOT_VALID);
 		return;
 	}
