diff --git a/tools/actions/initializer.py b/tools/actions/initializer.py
index 41a0be3..273cb46 100644
--- a/tools/actions/initializer.py
+++ b/tools/actions/initializer.py
@@ -129,6 +129,7 @@ def init(args):
         if not os.path.isdir(tools.config.defaults["overlay"]):
             os.mkdir(tools.config.defaults["overlay"])
             os.mkdir(tools.config.defaults["overlay"]+"/vendor")
+            os.mkdir(tools.config.defaults["overlay"]+"/nix")
         if not os.path.isdir(tools.config.defaults["overlay_rw"]):
             os.mkdir(tools.config.defaults["overlay_rw"])
             os.mkdir(tools.config.defaults["overlay_rw"]+"/system")
diff --git a/tools/helpers/lxc.py b/tools/helpers/lxc.py
index 6be7b7d..e23d17a 100644
--- a/tools/helpers/lxc.py
+++ b/tools/helpers/lxc.py
@@ -119,6 +119,10 @@ def generate_nodes_lxc_config(args):
     # NFC config
     make_entry("/system/etc/libnfc-nci.conf", options="bind,optional 0 0")
 
+    # Nix
+    make_entry("/nix", options="ro,bind 0 0")
+    make_entry("/run/binfmt", options="ro,bind,create=dir,optional 0 0")
+
     return nodes
 
 LXC_APPARMOR_PROFILE = "lxc-waydroid"
