diff --git a/src/hydra-queue-runner/state.hh b/src/hydra-queue-runner/state.hh
index 5a5c5f94..af0c18b8 100644
--- a/src/hydra-queue-runner/state.hh
+++ b/src/hydra-queue-runner/state.hh
@@ -272,7 +272,8 @@ struct Machine
            preferLocalBuild steps. */
         for (auto & f : mandatoryFeatures)
             if (!step->requiredSystemFeatures.count(f)
-                && !(f == "local" && step->preferLocalBuild))
+                && !(f == "local" && step->preferLocalBuild)
+                && !(f == "non-local" && !step->preferLocalBuild))
                 return false;
 
         /* Check that the machine supports all features required by
